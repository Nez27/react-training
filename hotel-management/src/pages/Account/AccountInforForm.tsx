import { useForm } from 'react-hook-form';

// Styled
import Input from '@src/commons/styles/Input';

// Components
import Form from '@src/components/Form';

// Constants
import { REQUIRED_FIELD_ERROR } from '@src/constants/formValidateMessage';

// Hooks
import { useUpdateAccount } from '@src/hooks/authentication/useUpdateAccount';
import { useAccount } from '@src/hooks/authentication/useAccount';

interface IAccountForm {
  fullName: string;
}

const AccountInforForm = () => {
  const { account } = useAccount();

  const { updateAccount, isUpdating } = useUpdateAccount();
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<IAccountForm>();

  const onSubmit = (data: IAccountForm) => {
    updateAccount(data);
  };

  return (
    <Form onSubmit={handleSubmit(onSubmit)}>
      <Form.Row label="Email:">
        <Input id="email" readOnly value={account?.email} />
      </Form.Row>

      <Form.Row label="Full Name:" error={errors?.fullName?.message as string}>
        <Input
          id="fullName"
          {...register('fullName', {
            required: REQUIRED_FIELD_ERROR,
            value: account?.user_metadata.fullName,
          })}
        />
      </Form.Row>

      <Form.Action>
        <Form.Button type="submit" name="submit" disabled={isUpdating}>
          Update account
        </Form.Button>
      </Form.Action>
    </Form>
  );
};

export default AccountInforForm;
