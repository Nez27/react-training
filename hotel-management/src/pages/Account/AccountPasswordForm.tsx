import { useEffect } from 'react';
import { useForm } from 'react-hook-form';
import toast from 'react-hot-toast';

// Styled
import Input from '@src/commons/styles/Input';

// Components
import Form from '@src/components/Form';

// Constants
import { REGEX } from '@src/constants/commons';
import {
  PASSWORD_CONDITION,
  PASSWORD_NOT_MATCH,
  PASSWORD_NOT_MATCH_REQUIREMENT,
} from '@src/constants/formValidateMessage';

// Helpers
import { isValidRegex } from '@src/helpers/validators';

// Hooks
import { useUpdateAccount } from '@src/hooks/authentication/useUpdateAccount';

interface IAccountPasswordForm {
  password: string;
  confirmPassword: string;
}

const AccountPasswordForm = () => {
  const { updateAccount, isUpdating } = useUpdateAccount();
  const {
    register,
    handleSubmit,
    getValues,
    reset,
    formState: { errors },
  } = useForm<IAccountPasswordForm>();

  useEffect(() => {
    if (errors?.password?.message) {
      toast.error(PASSWORD_CONDITION);
    }
  }, [errors?.password?.message]);

  const onSubmit = (data: IAccountPasswordForm) => {
    updateAccount(data, {
      onSuccess: () => {
        reset();
      },
    });
  };

  return (
    <Form onSubmit={handleSubmit(onSubmit)}>
      <Form.Row label="Password:" error={errors?.password?.message}>
        <Input
          id="password"
          type="password"
          {...register('password', {
            validate: {
              checkPassword: (v) =>
                isValidRegex(REGEX.PASSWORD, v) ||
                PASSWORD_NOT_MATCH_REQUIREMENT,
            },
          })}
        />
      </Form.Row>

      <Form.Row
        label="Confirm Password:"
        error={errors?.confirmPassword?.message}
      >
        <Input
          id="confirmPassword"
          type="password"
          {...register('confirmPassword', {
            validate: (value) =>
              getValues().password === value || PASSWORD_NOT_MATCH,
          })}
        />
      </Form.Row>

      <Form.Action>
        <Form.Button type="submit" name="submit" disabled={isUpdating}>
          Update password
        </Form.Button>
      </Form.Action>
    </Form>
  );
};

export default AccountPasswordForm;
