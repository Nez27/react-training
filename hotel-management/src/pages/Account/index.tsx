// Styled
import { FormArea, StyledAccount, SubTitle, Title } from './styled';

// Components
import AccountInforForm from './AccountInforForm';
import AccountPasswordForm from './AccountPasswordForm';

const Account = () => {
  return (
    <StyledAccount>
      <Title>Update account</Title>

      <FormArea>
        <SubTitle>Update information</SubTitle>
        <AccountInforForm />
      </FormArea>

      <FormArea>
        <SubTitle>Update password</SubTitle>
        <AccountPasswordForm />
      </FormArea>
    </StyledAccount>
  );
};

export default Account;
