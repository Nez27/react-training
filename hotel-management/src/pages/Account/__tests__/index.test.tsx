import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

// Components
import Account from '..';

describe('Account', () => {
  test('Should render correctly', () => {
    const queryClient = new QueryClient();
    const wrapper = render(
      <QueryClientProvider client={queryClient}>
        <Account />
      </QueryClientProvider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
