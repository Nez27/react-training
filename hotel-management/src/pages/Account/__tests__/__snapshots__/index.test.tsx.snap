// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-dlniIP jaPwYJ"
      >
        <h2
          class="sc-bdnyFh gafMEQ"
        >
          Update account
        </h2>
        <div
          class="sc-hKFymg iUYoZl"
        >
          <h3
            class="sc-gtsqUy fkdaKz"
          >
            Update information
          </h3>
          <form
            class="sc-gKAaef fefJbB"
          >
            <div
              class="sc-fujznN ioyvGl"
              direction="horizontal"
            >
              <label
                class="sc-pNWRh gSLqll"
              >
                Email:
              </label>
              <div>
                <input
                  class="sc-eCAqax bNAfHQ"
                  id="email"
                  readonly=""
                  type="text"
                />
              </div>
            </div>
            <div
              class="sc-fujznN ioyvGl"
              direction="horizontal"
            >
              <label
                class="sc-pNWRh gSLqll"
              >
                Full Name:
              </label>
              <div>
                <input
                  class="sc-eCAqax bNAfHQ"
                  id="fullName"
                  name="fullName"
                  type="text"
                />
              </div>
            </div>
            <div
              class="sc-iCoHzw jMTsfU"
            >
              <button
                class="sc-jSFipO sc-kEqXeH fQJaue bcOLTW"
                name="submit"
                type="submit"
                variations="primary"
              >
                Update account
              </button>
            </div>
          </form>
        </div>
        <div
          class="sc-hKFymg iUYoZl"
        >
          <h3
            class="sc-gtsqUy fkdaKz"
          >
            Update password
          </h3>
          <form
            class="sc-gKAaef fefJbB"
          >
            <div
              class="sc-fujznN ioyvGl"
              direction="horizontal"
            >
              <label
                class="sc-pNWRh gSLqll"
              >
                Password:
              </label>
              <div>
                <input
                  class="sc-eCAqax bNAfHQ"
                  id="password"
                  name="password"
                  type="password"
                />
              </div>
            </div>
            <div
              class="sc-fujznN ioyvGl"
              direction="horizontal"
            >
              <label
                class="sc-pNWRh gSLqll"
              >
                Confirm Password:
              </label>
              <div>
                <input
                  class="sc-eCAqax bNAfHQ"
                  id="confirmPassword"
                  name="confirmPassword"
                  type="password"
                />
              </div>
            </div>
            <div
              class="sc-iCoHzw jMTsfU"
            >
              <button
                class="sc-jSFipO sc-kEqXeH fQJaue bcOLTW"
                name="submit"
                type="submit"
                variations="primary"
              >
                Update password
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-dlniIP jaPwYJ"
    >
      <h2
        class="sc-bdnyFh gafMEQ"
      >
        Update account
      </h2>
      <div
        class="sc-hKFymg iUYoZl"
      >
        <h3
          class="sc-gtsqUy fkdaKz"
        >
          Update information
        </h3>
        <form
          class="sc-gKAaef fefJbB"
        >
          <div
            class="sc-fujznN ioyvGl"
            direction="horizontal"
          >
            <label
              class="sc-pNWRh gSLqll"
            >
              Email:
            </label>
            <div>
              <input
                class="sc-eCAqax bNAfHQ"
                id="email"
                readonly=""
                type="text"
              />
            </div>
          </div>
          <div
            class="sc-fujznN ioyvGl"
            direction="horizontal"
          >
            <label
              class="sc-pNWRh gSLqll"
            >
              Full Name:
            </label>
            <div>
              <input
                class="sc-eCAqax bNAfHQ"
                id="fullName"
                name="fullName"
                type="text"
              />
            </div>
          </div>
          <div
            class="sc-iCoHzw jMTsfU"
          >
            <button
              class="sc-jSFipO sc-kEqXeH fQJaue bcOLTW"
              name="submit"
              type="submit"
              variations="primary"
            >
              Update account
            </button>
          </div>
        </form>
      </div>
      <div
        class="sc-hKFymg iUYoZl"
      >
        <h3
          class="sc-gtsqUy fkdaKz"
        >
          Update password
        </h3>
        <form
          class="sc-gKAaef fefJbB"
        >
          <div
            class="sc-fujznN ioyvGl"
            direction="horizontal"
          >
            <label
              class="sc-pNWRh gSLqll"
            >
              Password:
            </label>
            <div>
              <input
                class="sc-eCAqax bNAfHQ"
                id="password"
                name="password"
                type="password"
              />
            </div>
          </div>
          <div
            class="sc-fujznN ioyvGl"
            direction="horizontal"
          >
            <label
              class="sc-pNWRh gSLqll"
            >
              Confirm Password:
            </label>
            <div>
              <input
                class="sc-eCAqax bNAfHQ"
                id="confirmPassword"
                name="confirmPassword"
                type="password"
              />
            </div>
          </div>
          <div
            class="sc-iCoHzw jMTsfU"
          >
            <button
              class="sc-jSFipO sc-kEqXeH fQJaue bcOLTW"
              name="submit"
              type="submit"
              variations="primary"
            >
              Update password
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
